packages/core/admin/admin/src/pages/SettingsPage/pages/AuditLogs/ListView/ModalDialog/index.js

```diff
@@ -1,29 +1,60 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
 import { ModalLayout, ModalHeader, ModalBody } from '@strapi/design-system/ModalLayout';
 import { Breadcrumbs, Crumb } from '@strapi/design-system/Breadcrumbs';
+import { Loader } from '@strapi/design-system/Loader';
 import { Grid } from '@strapi/design-system/Grid';
 import { Box } from '@strapi/design-system/Box';
+import { Flex } from '@strapi/design-system/Flex';
 import { Typography } from '@strapi/design-system/Typography';
-import { pxToRem } from '@strapi/helper-plugin';
+import { pxToRem, useNotification } from '@strapi/helper-plugin';
 import getDefaultMessage from '../utils/getActionTypesDefaultMessages';
 import useFormatTimeStamp from '../hooks/useFormatTimeStamp';
 import ActionItem from './ActionItem';
+import { useFetchClient } from '../../../../../../hooks';
+
+const ModalDialog = ({ onClose, logId }) => {
+  const { get } = useFetchClient();
+  const toggleNotification = useNotification();
+
+  const fetchAuditLog = async (id) => {
+    const { data } = await get(`/admin/audit-logs/${id}`);
+
+    return data;
+  };
+
+  const { data: { date, user, action, payload } = {}, status } = useQuery(
+    ['audit-log', logId],
+    () => fetchAuditLog(logId),
+    {
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'notification.error', defaultMessage: 'An error occured' },
+        });
+        onClose();
+      },
+    }
+  );
 
-const ModalDialog = ({ onToggle, data: { date, user, action } }) => {
   const { formatMessage } = useIntl();
   const formatTimeStamp = useFormatTimeStamp();
-  const formattedDate = formatTimeStamp(date);
 
-  return (
-    <ModalLayout onClose={onToggle} labelledBy="title">
-      <ModalHeader>
-        <Breadcrumbs label={formattedDate}>
-          <Crumb>{formattedDate}</Crumb>
-        </Breadcrumbs>
-      </ModalHeader>
-      <ModalBody>
+  const formattedDate = status === 'success' ? formatTimeStamp(date) : '';
```

Comment:
Shouldn't we just handle the error before formatting the date?

Reply:
I think when errors happen it shouldn't get to this point because I close the modal the error event

The check here was to make sure the loading state doesn't make it crash

Reply:
So could you not put the variable definition after the loading state is handled and avoid this ternary?

Reply:
Yeah that's what we usually do, but I don't think I can here. Because I'm using the variable in ModalHeader, but I'm managing the loading state inside ModalBody (because I want to keep the same modal instance across all statuses, and avoid recreating a modal on each change)




I'm handling the loading state inside ModalBody,

Reply:
oh gotcha, didn't notice the loading state was inside the `getModalBody`, works for me 👍